frontend/public/components/build-pipeline.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Link } from 'react-router-dom';
+import { resourcePath } from './utils';
+import { fromNow } from './utils/datetime';
+
+const getBuildNumber = (resource: any) => _.get(resource, ['metadata', 'annotations', 'openshift.io/build.number']);
+const getStages = (status: any) => status && status.stages || [];
+const getJenkinsStatus = (resource: any) => {
+  const status = _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-status-json']);
+  return status && JSON.parse(status);
+};
+const getJenkinsLogURL = (resource: any) => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-log-url']);
+const getJenkinsBuildURL = (resource: any) => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-build-uri']); 
+
+
+const BuildSummaryStatusIcon: React.SFC<BuildSummaryStatusIconProps> = ({status}) => {
+  let statusClass = _.lowerCase(status);
+  let icon = ({
+    new: 'fa-hourglass-o',
+    pending: 'fa-hourglass-half',
+    running: 'fa-refresh fa-spin',
+    complete: 'fa-check-circle',
+    failed: 'fa-times-circle'
+  })[statusClass];
+  let hide = icon ? 'hide' : '';
+  console.log('status is: ', status, statusClass);
+  return <React.Fragment>
+    <span className={`build-pipeline__status-icon build-pipeline__status-icon--${statusClass}`}>
```

Comment:
Would this be to conditionally add the `build-pipeline__status-icon--${status}` class, ONLY if there is a valid status?  If I could do this:

```javascript
  const klass = classNames(
    'build-pipeline__status-icon',
    {`build-pipeline__status-icon--${statusClass}` : !!statusClass}
  );
```
I think it would make sense, but the ``backtick`` operator isn't valid for creating an object key, so it doesn't help me take advantage of the conditional logic provided by `className`.